device/globals/exit.gd

```diff
@@ -0,0 +1,44 @@
+extends TextureButton  # XXX: Use TextureButton until background is fixed so CollisionPolygon2D will work
+
+export(String, FILE, ".esc") var esc_script  # must contain :dblclick
```

Comment:
What is the purpose of `:dblclick`? Wouldn't it make more sense to have a project setting for making double-click on exit nodes teleport the player to the exit and have a single `:exit` action or something trigger other events, whether the player fast-travelled to the exit or just walked over?

Reply:
Especially if exit were replaced with hotspot, having scriptable actions makes sense to me.

I may have stated this somewhere, but if there was a single teleport, your game couldn't implement the scenario where the player tries to leave a screen and instead says "I'm not finished here yet." or something. At least not easily.

Reply:
Wouldn't you have state variables that you set when each of the goals of the scene are achieved anyway though, like if you have a certain item in your inventory (which might again have other dependencies?

If `:dblclick` is a separate event, you need to do the same checks in both `:exit` and `:dblclick`